gtsfm/bundle/bundle_adjustment.py

```diff
@@ -61,12 +63,55 @@
 logger = logger_utils.get_logger()
 
 
-class BundleAdjustmentOptimizer:
+class TwoViewBundleAdjustmentProcess(GTSFMProcess):
```

Comment:
Should be called TwoViewBundleAdjustment and be in its own file I think

Reply:
The issue with putting it in a separate file is that since that file is not imported anywhere, the registry never sees the dummy class. If I import it somewhere but don't use the dummy class, the linter will complain of an unused import.

Reply:
I’m confused, if it’s never used, why is it in our graph? Isn’t the fact that it is in our graph evidence for the fact that it is used?

Reply:
Both two-view and global bundle adjustment are done via the BundleAdjustmentOptimizer class, but there's no way to show that in the graph as currently constructed. Every process shown as a blue node must be a unique subclass of GTSFMProcess. So Akshay and I added the dummy class TwoViewBundleAdjustment to get a separate UiMetadata object. 

The reason this works is because the `bundle_adjustment.py` file is imported and used when running GTSFM, meaning both TwoViewBundleAdjustment and BundleAdjustmentOptimizer are visible to the central registry. Earlier I tried creating a separate file for TwoViewBundleAdjustment, but that renders it invisible to the registry unless the file is imported somewhere. However, importing the file somewhere will trigger a flake8 error (`F401 'gtsfm.bundle.two_view_ba' imported but unused`) because the dummy class in the file doesn't have any use besides exposing TwoViewBA-related nodes to the registry.

Reply:
TLDR: a class with the same name cannot be used in 2 places in the graph (due to the registration pattern), so we had to create a dummy one for the 2-view BA.

Reply:
^Its hacky and Im not super happy with it, open to suggestions :)

Reply:
It seems to me it does not have to be dummy: you can have 2 derived classes from one base class that does not have the meta-data: 
BundleAdjustment (everything except metadata)
- TwoViewBundleAdjustment (just 2 view metadata)
- GlobalBundleAdjustment (just global metadata)
As per style guide, every class should be in its own file.

Reply:
Done here: https://github.com/borglab/gtsfm/pull/540/commits/15dfd7849433ca0018e97340afe5e1386ac9804c